features/storage/csi.feature

```diff
@@ -197,4 +197,81 @@ Feature: CSI testing related feature
+    Given I have a project
+    Given I obtain test data file "storage/misc/pvc.json"
+    When I create a dynamic pvc from "pvc.json" replacing paths:
+      | ["metadata"]["name"]         | mypvc     |
+      | ["spec"]["storageClassName"] | <sc_name> |
+      | ["spec"]["volumeMode"]       | Block     |
+    Then the step should succeed
+    Given I obtain test data file "storage/misc/pod-with-block-volume.yaml"
+    When I run oc create over "pod-with-block-volume.yaml" replacing paths:
+      | ["metadata"]["name"]                                         | mypod       |
+      | ["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc       |
+      | ["spec"]["containers"][0]["volumeDevices"][0]["devicePath"]  | /dev/dblock |
+    Then the step should succeed
+    And the pod named "mypod" becomes ready
+    And the "mypvc" PVC becomes :bound
+    When I execute on the pod:
+      | sh | -c | [[ -b /dev/dblock ]] |
+    Then the step should succeed
+    When I execute on the pod:
+      | /bin/dd | if=/dev/zero | of=/dev/dblock | bs=1M | count=1 |
+    Then the step should succeed
+    When I execute on the pod:
+      | sh | -c | echo "test data" > /dev/dblock |
+    Then the step should succeed
+    When I execute on the pod:
+      | /bin/dd | if=/dev/dblock | of=/tmp/testfile | bs=1M | count=1 |
+    Then the step should succeed
+    When I execute on the pod:
+      | sh | -c | cat /tmp/testfile |
+    Then the step should succeed
+    And the output should contain "test data"
+
+    Examples:
+      | sc_name      |
+      | standard-csi | # @case_id OCP-37564
+
+
+  # @author wduan@redhat.com
+  @admin
+  Scenario Outline: CSI dynamic provisioning with different type
+    Given I have a project
+    And admin clones storage class "sc-<%= project.name %>" from "<sc_name>" with:
+      | ["parameters"]["type"] | <type> |
 
+    Given I obtain test data file "storage/misc/pvc.json"
+    When I create a dynamic pvc from "pvc.json" replacing paths:
+      | ["metadata"]["name"]                         | mypvc                  |
+      | ["spec"]["storageClassName"]                 | sc-<%= project.name %> |
+      | ["spec"]["resources"]["requests"]["storage"] | <size>                 |
+    Then the step should succeed
+
+    Given I obtain test data file "storage/misc/deployment.yaml"
+    When I run oc create over "deployment.yaml" replacing paths:
+      | ["metadata"]["name"]                                                             | mydep      |
+      | ["spec"]["template"]["metadata"]["labels"]["action"]                             | storage    |
+      | ["spec"]["template"]["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc      |
+      | ["spec"]["template"]["spec"]["containers"][0]["volumeMounts"][0]["mountPath"]    | /mnt/local |
+    Then the step should succeed
+    And a pod becomes ready with labels:
+      | action=storage |
+    And the "mypvc" PVC becomes :bound
```

Comment:
Suggest to check pv volume size and volume type

Reply:
@chao007, thanks for the suggestion, seems the volume type like sc/st1 doesn't exist in the pvc or pv into. I added the check for pvc size and passed. 

>         And the "mypvc" PVC becomes :bound                                           # features/step_definitions/pvc.rb:3
>       [03:10:08] INFO> 
>       [03:10:08] INFO> After 1 iterations and 2 seconds:
>       matched status for persistentvolumeclaims mypvc: 'bound' while expecting '[:bound]'
>         And the expression should be true> pvc.capacity == "125Gi"                   # features/step_definitions/common.rb:133
> waiting for operation up to 3600 seconds..
>         When I execute on the pod:                                                   # features/step_definitions/pod.rb:216
>           | sh | -c | echo "test" > /mnt/local/testfile |